userprofile.py

```diff
@@ -18,9 +18,9 @@ class UserProfile:
 	def __init__(self):
 		self.pokemon = []
 
-	def user(self) -> discord.User:
+	def user(self, user_id: Optional[int]) -> discord.User:
 		"""Discord user associated with this profile."""
-		pass
+		self.user_id = user_id
```

Comment:
This does not make any sense. It is supposed to **get** the `discord.User` associated with the profile, not set it. You will need a reference to the discord bot. Try to avoid providing `bot` as an argument.